File: 프로그래머스/0/120834. 외계행성의 나이/외계행성의 나이.java
class Solution {
    public String solution(int age) {
        String answer = "";
        int num = 0; 
        int le = 0; 
        
        if (age == 1000) {
            le = 4;
        } else if (age >= 100) {
            le = 3;
        } else if (age >= 10) {
            le = 2;
        } else if (age >= 1) {
            le = 1;
        }
        
        char []en = new char[26];
        
        for (int i = 0; i < 26; i++) {
            en[i] = (char)('a'+i);
        }
        
        for (int i = 0; i < le; i++) {
            num = age%10;
            age /= 10;
            
            answer = en[num] + answer;
        }
        return answer;
    }
}

File: 프로그래머스/0/120834. 외계행성의 나이/외계행성의 나이.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(int age) {
    int n = 0; 
    
    if (age >= 1000) {
        n = 4;
    } else if (age >= 100) {
        n = 3;
    } else if (age >= 10) {
        n = 2;
    } else {
        n = 1;
    }
    char* answer = (char*)malloc(sizeof(char)*n);
    for (int i = 0; i < n*2; i++) {
        answer[i] = '\0';
    }
    
    int le = n;
    int le_2 = n;
    n--;
    
    int num = age;
    
    for (int i = 0; i < le; i++) {
        num = age%10;
        age /= 10;
        answer[n] = (char)('a'+num);
        n--;
    }

    return answer;
}